### PR TITLE
Logout on token expiration

### PR DESCRIPTION
**Pull request by gberaudo**

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -4,32 +4,29 @@ const cognitoState = 'cognito_state';
...
+        this.setAccessToken(params.get('id_token'));
       }
-    } catch (e) {
-      // do nothing
     }
 
     const accessToken = this.getAccessToken();
     if (accessToken) {
-      _AWSCredentials = fromCognitoIdentityPool({
+      window['AWSCred'] = _AWSCredentials = fromCognitoIdentityPool({
```
Comment (fredj) [inline src/auth.js:29]:
```suggestion
      _AWSCredentials = fromCognitoIdentityPool({
```
I guess it was only for the debug

File: src/elements/ngm-auth.js [hunk 2]
```diff
@@ -25,12 +25,32 @@ class NgmAuth extends I18nMixin(LitElement) {
...
     this.redirectUri = `${location.origin}${location.pathname}`;
     this.scope = 'openid+profile';
   }
 
+  /**
+   * Set or clear auto logout timer.
+   * @param claims
+   */
+  updateLogoutTimeout_(claims) {
+    if (this.expireTimer_) {
```
Comment (fredj) [inline src/elements/ngm-auth.js:39]: will it work ? shouldn't `this.expireTimer_` be set to `null` after `clearTimeout(this.expireTimer_);` ?

File: src/auth.js [hunk 3]
```diff
@@ -4,32 +4,29 @@ const cognitoState = 'cognito_state';
...
-      if (response.token_type === 'Bearer' && response.state === this.state()) {
-        this.setUser(this.parseToken(response.access_token));
-        this.setAccessToken(response.id_token);
+    if (window.location.hash.startsWith('#')) {
+      // https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html
+      const response = window.location.hash.substring(1);
+      const params = new URLSearchParams(response);
+      if (params.has('access_token') && params.has('id_token') &&
+          params.get('token_type') === 'Bearer' && params.get('state') === this.state()) {
+        localStorage.setItem('rawCognitoResponse', response);
```
Comment (fredj) [inline src/auth.js:18]: minor but `localStorage.removeItem('rawCognitoResponse');` could be added in `logout`

Comment (fredj) [reply src/auth.js:18]: but the item is never read, remove this line ?

File: src/auth.js (hunk 1 above)
Comment (gberaudo) [reply src/auth.js:29]: Andrea said he did it on purpose in its code. So I restored it.

File: src/auth.js (hunk 3 above)
Comment (gberaudo) [reply src/auth.js:18]: It is for debugging. The login process is very complicate and envolve opening and closing browser windows.
As a consequence it is not easy to see the cognito response (breakpoints do not work) when debugging, except by storing it in the local storage.

**Review by fredj — APPROVED** [review]

**Pull request merged**